File: Java/Examples4/ThreeDimensionalObjectFactory.java
public class ThreeDimensionalObjectFactory {

    public static String getMenu() {
	String menu = " 1. 球\n" +
	              " 2. 正八面体";
	return menu;
    }

    public static IThreeDimensionalObject create(String name) 
	throws VolumeCalculatorException {
	if( name.equals("1") || name.equals("球") ) {    		
	    return new Sphere();
    	} else if( name.equals("2") || name.equals("正八面体") ) {
	    return new RegularOctahedron();    	
    	} else {    		
	    throw new VolumeCalculatorException(
		   name + " は無効です．" );
	}
    }
}

File: Python/Examples2/three_dimensional_object_factory.py
from volume_calculator_exception import VolumeCalculatorException
from sphere import Sphere
from cylinder import Cylinder
from cone import Cone

class ThreeDimensionalObjectFactory:

    obj_list = [ 
                Sphere(), 
                Cylinder(),
                Cone()
                ]
            
    @classmethod
    def get_menu(cls): 
        menu = ''
        for idx in range(len(cls.obj_list)):
            menu += '{0}.\t {1}\n'.format(idx+1,cls.obj_list[idx].name)        
        return menu
        
        
    @classmethod
    def create(cls,name):
        if name == '1' or name == Sphere().name:
            return Sphere()
        elif name == '2' or name == Cylinder().name:
            return Cylinder()
        elif name == '3' or name == Cone().name:
            return Cone()            
        else:
            raise VolumeCalculatorException(name  + ' は無効です．')
        
